packages/@aws-cdk/aws-glue-alpha/lib/job.ts

```diff
@@ -719,6 +719,16 @@ export class Job extends JobBase {
       }
     }
 
+    if (props.maxCapacity !== undefined && (props.workerType && props.workerCount !== undefined)) {
```

Comment:
Sorry for not thinking of this earlier, but I have one more thing I think we should add. There is a small chance that `maxCapacity` and `workerCount` could be `Tokens`. Could we wrap this validation in an if statement that first checks if both `maxCapacity` and `workerCount` are resolved. As an example:

```ts
if (!Token.isUnresolved(props.maxCapacity) && !Token.isUnresolved(props.workerCount)) {
  // your code here
}
```

We should probably also include an associated unit test for this as well. I hope this makes sense. Let me know if you have any questions.

Reply:
@colifran 

> There is a small chance that maxCapacity and workerCount could be Tokens.

I see. However, I assume that `props. maxCapacity` and `props. workerCount` are input values from the user, not values generated from the resource, but when do they become Unresolved? Depending on that, I will consider new unit tests.

Reply:
@go-to-k I realize it's a small chance, but it's better to add the check for it now since we're adding new code rather than having to go back and fix a bug. Basically, this would only occur if someone were to provide those as `Lazy` values. As an example:

```ts
{
  workerCount: cdk.Lazy.number({ produce: () => 5 }),
  maxCapacity: cdk.Lazy.number({ produce: () => 5 }),
}
```

Reply:
Or when using a `Parameter` it can also happen.

Reply:
@colifran 

Thanks, I understand.

In the following code, when `workerCount` and `maxCapacity` are `Token`, it does not validate.

```ts
if (!Token.isUnresolved(props.maxCapacity) && !Token.isUnresolved(props.workerCount)) {
  // your code here
}
```

However, even if the value is a `Token` and `Unresolved`, the fact that you are specifying that combination of parameters is itself a problem (i.e., specifying both `workerCount` and `maxCapacity`, regardless of the value, is itself a problem).

Therefore, shouldn't they be validated whether they are tokens or not?

I think we can check if the parameters are undefined or not, even if encoded numbers as `Token`.

Reply:
@go-to-k good catch. I agree with this. We're checking for just the existence of the properties and not actually validating the value of the properties. That said, I think this is good to go. Thanks for your effort on this!